src/Common/src/TypeSystem/CodeGen/TypeDesc.CodeGen.cs

```diff
@@ -0,0 +1,50 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Internal.TypeSystem
+{
+    // Extensions to TypeDesc suitable for code generation purposes
+    partial class TypeDesc
+    {
+        /// <summary>
+        /// Validates that a type can be fully loaded. Throws an exception if a failure occurs.
+        /// </summary>
+        public virtual void ValidateCanLoad()
```

Comment:
> I do not think that this validation should be on TypeDesc

TypeDesc doesn't expose proper hooks that would let us do the validation without awkward-looking code. (E.g. I want to make sure layout can be computed... all I can do is access e.g. InstanceSize property and make a dead store of it's value somewhere).

Are you suggesting we should expose `EnsureInstanceLayout`, `EnsureStaticLayout`, `EnsureRuntimeInterfaces`, etc. on `TypeDesc` instead?

> validation should be on EETypeNode instead

RyuJIT only asks for the R2R helper. By the time we're getting the associated EEType node, we already forgot what method wanted it. At that point all we can do is tear down the process which is the same as not doing any validation at all...